icgcget/cli.py

```diff
@@ -79,123 +95,211 @@ def cli(ctx, config, logfile):
-
-    elif repo == 'gdc':
-        if manifest is True:
-
-            code = gdc_client.gdc_manifest_call(fileid, gdc_access, gdc_path, output, gdc_udt,
-                                                gdc_transport_parallel)
-        elif fileid is not None:
-            code = gdc_client.gdc_call(fileid, gdc_access, gdc_path, output, gdc_udt, gdc_transport_parallel)
-        if code != 0:
-            logger.error(repo + " exited with a nonzero error code.")
-
-    return code
+            check_access(None, 'ega')
+        if ega_transport_parallel != '1':
+            logger.warning("Parallel streams on the ega client may cause reliability issues and failed " +
+                           "downloads.  This option is not recommended.")
+        code = ega_client.ega_call(object_ids['ega'], ega_username, ega_password, ega_path,
+                                   ega_transport_parallel, ega_udt, output)
+        check_code('Ega', code)
 
+    if 'collaboratory' in object_ids and object_ids['collaboratory']:
+        check_access(icgc_access, 'icgc')
+        code = icgc_client.icgc_manifest_call(object_ids['collaboratory'], icgc_access, icgc_path,
+                                              icgc_transport_file_from, icgc_transport_parallel, output, 'collab')
+        check_code('Icgc', code)
 
-if __name__ == "__main__":
+    if 'gdc' in object_ids and object_ids['gdc']:
+        check_access(gdc_access, 'gdc')
+        code = gdc_client.gdc_manifest_call(object_ids['gdc'], gdc_access, gdc_path, output, gdc_udt,
+                                            gdc_transport_parallel)
+        check_code('Gdc', code)
+
+
+@cli.command()
+@click.argument('fileids', nargs=-1, required=True)
+@click.option('--repos', '-r', type=click.Choice(REPOS),  multiple=True)
+@click.option('--manifest', '-m', is_flag=True, default=False)
+@click.option('--cghub-access', type=click.STRING)
+@click.option('--ega-username', type=click.STRING)
+@click.option('--ega-password', type=click.STRING)
+@click.option('--gdc-access', type=click.STRING)
+@click.option('--icgc-access', type=click.STRING)
+@click.pass_context
+def dryrun(ctx, repos, fileids, manifest, cghub_access, ega_username, ega_password, gdc_access, icgc_access):
+
+    repo_list = []
+    gdc_ids = []
+    if os.getenv("ICGCGET_API_URL"):
+        api_url = os.getenv("ICGCGET_API_URL")
+    else:
+        api_url = ctx.default_map["portal_url"] + ctx.default_map["portal_api"]
+    total_size = 0
+    table = [["", "Size", "Unit", "File Format", "Repo"]]
+
+    if manifest:
+        if len(fileids) > 1:
+            logger.warning("For download from manifest files, multiple manifest id arguments is not supported")
+            raise click.BadArgumentUsage("Multiple manifest files specified.")
+        try:
+            manifest_json = icgc_api.get_manifest_id(fileids[0], api_url, repos)
+        except RuntimeError:
+            raise click.Abort
```

Comment:
For all of the methods that call an api, if there is an error, the user will see the output of the common call_api method.  If the api did not receive a response for whatever reason it raises a runtime exception, which gets caught silently by upper layers of code.  If it receives an error response it raises an HTTPError, which is used by middle layers of code to see if the response was an expected type, and is then handled there and a runtime exception is raised.